www/source/blog/2018-10-12-shelving-composites.html.md

```diff
@@ -0,0 +1,22 @@
+---
+title: Shelving Composites (for now)
+date: 2018-10-12
+author: Garrett Amini
+tags: supervisor, packaging
+category: supervisor
+classes: body-article
+---
+
+Nearly exactly one year ago today, Christopher Maier [presented the ability to create _composite packages_](https://www.habitat.sh/blog/2017/10/Introducing-Composites/), a special kind of Habitat package that includes other Habitat packages in order to be managed by a single Supervisor. The Habitat community had been finding all sorts of creative ways to use Habitat to incorporate their services under a single Supervisor, so we sought a way to canonically enable them to do so. 
+
+This past year has taught us that while we were on the right track with the idea of composite packages, our implementation left much to be desired. Users kept trying to utilize composite packages, but would inevitably find themselves bumping up against limitations of the feature, and would thus end implementing the same sorts of creative solutions we were trying to simplify in the first place.
+
+## What's Happening Now?
+
+Starting with version *X.XX.X*, we will be removing the ability to run composite plans from Habitat. We've learned enough over the past year to realize that we need a complete re-implementation of how that feature will work with the Supervisor, so we don't want to lead our users down a dead-end path. This is one of those cases where rebuilding the feature will be much faster to implement than refactoring what we have, so we are taking the hard step of removing the current implementation now.

```

Comment:
Rather than say it's easier to rebuild rather than refactor (which is often a rash decision), I'd frame this more as "we put out a prototype and have learned a lot from it; we're now going to make the real feature". 

With that knowledge, we're going to take out the existing code, which should simplify our current stability and push-to-1.0 work.

Reply:
It would also be good to really emphasize that the code is _not being removed right now_, and there will be ample time for any users of composites to consider their next moves.